pg_search/src/index/state.rs

```diff
@@ -33,6 +35,35 @@ pub struct SearchState {
     pub schema: SearchIndexSchema,
 }
 
+#[derive(Debug, Clone)]
+pub struct SearchHit {
+    pub score: f32,
+    pub ctid: u64,
+    pub doc_address: Option<DocAddress>,
+    pub key: Option<TantivyValue>,
+}
+
+impl Eq for SearchHit {}
+
+impl PartialEq<Self> for SearchHit {
+    #[inline(always)]
+    fn eq(&self, other: &Self) -> bool {
+        self.score == other.score && self.key == other.key
+    }
+}
+
+impl PartialOrd for SearchHit {
+    #[inline(always)]
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        match self.score.partial_cmp(&other.score) {
+            Some(Ordering::Equal) => self.key.partial_cmp(&other.key),
```

Comment:
I feel like the comparison for `key`, if it's necessary, should be `.reverse()`'d, such that docs with the same score will be subsorted by their key in ASCending order.  As it stands now, they end up in DESCending order due to Tantivy sorting docs by score in descending order.

Reply:
yeah, I was right.  This is what causes some tests to light up.